### PR TITLE
CI: caching tune (part one - relocate playwright caching)

### PR DESCRIPTION
**Pull request by kalessil**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

In https://github.com/woocommerce/woocommerce/pull/49020 we further tuning up deps caching, and this PR extracts Playwright download caching. The produced artifact does not depend on the target package the action requested to install.

Relocating the caching in order to:
- execute it based on action inputs
- combine multiple caching actions to reduce the action invocation time

### Testing

CI checks shall pass and the cache should be fetched during monorepo install - see [here](https://github.com/woocommerce/woocommerce/actions/runs/9772948515/job/26978251045?pr=49081#step:3:133).

Screenshot from 2024-07-03 08-37-24

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -119,6 +119,7 @@ jobs:
         with:
           install: '${{ matrix.projectName }}...'
           build: 'false'
+          pull-playwright-cache: ${{ matrix.testEnv.shouldCreate }}
```
Comment (adimoldovan) [inline .github/workflows/ci.yml]: An improvement here would be to only run for e2e tests, they're the only ones using the browsers. We have the `matrix.testType` for that.

```suggestion
          pull-playwright-cache: ${{ matrix.testType == 'e2e' }}
```

Comment (kalessil) [reply .github/workflows/ci.yml]: Nice! But when we spinoff test envs, Playwright installs chromium for both unti/e2e tests Here example for a unit test job:

Screenshot from 2024-07-03 13-58-34

Comment (kalessil) [reply .github/workflows/ci.yml]: I'll try to fix using env:dev vs env:test for running unit and e2e tests accordingly (in this PR).

Comment (kalessil) [reply .github/workflows/ci.yml]: Let see how it works: 9c21959fde82fd6d064592ce1ec6d4619d4237bb

File: plugins/woocommerce/composer.lock [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
     "_readme": [
-        "This file locks the dependencies of your project to a known state",
+        "This file locks the dependencies of your project to a known state.",
```
Comment (kalessil) [inline plugins/woocommerce/composer.lock]: To be reverted before merge: only triggers CI job.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (adimoldovan) [reply .github/workflows/ci.yml]: Ah, you're right, I spoke too soon. I fixed that spinoff test envs part in [another PR](https://github.com/woocommerce/woocommerce/pull/49017), but it's not yet merged. I basically moved the playwright install from the env setup to the test run command.
But this switch to `env:dev` will also help.

Comment (kalessil) [reply .github/workflows/ci.yml]: Great! With that perspective, I think I need to revert my changes switching to `env:dev` - `env:test` reads as a better fit.

Comment (adimoldovan) [reply .github/workflows/ci.yml]: Your call. `env:dev` will work just as well, but I agree `env:test` reads better. This reminds me that we really need to audit and clean these pnpm scripts...

**Comment by kalessil** [comment]
Note: complimentary changes around Playwrite are in https://github.com/woocommerce/woocommerce/pull/49017 (Playwrite downloads will be triggered in E2E environments).

**Pull request merged**